Base/ctkAppLauncherEnvironment.h

```diff
@@ -48,6 +48,38 @@ class ctkAppLauncherEnvironment : public QObject
   /// using the `APPLAUNCHER_<level>_<varname>` convention.
   ///
   /// The variables matching `APPLAUNCHER_<level+1>_<varname>` are not returned.
+  ///
+  /// For example, assuming the current environment contains the following
+  /// variables:
+  ///
+  /// \code
+  /// APPLAUNCHER_LEVEL=2
+  /// APPLAUNCHER_0_PATH=/path/to/level-0
+  /// APPLAUNCHER_0_FRUIT=apple
+  /// APPLAUNCHER_1_PATH=/path/to/level-1:/path/to/level-0
+  /// APPLAUNCHER_1_FRUIT=banana
+  /// PATH=/path/to/level-2:/path/to/level-1:/path/to/level-0
+  /// FRUIT=kiwi
+  /// \endcode
+  ///
+  /// calling `environment(1)` would return:
+  ///
+  /// \code
+  /// APPLAUNCHER_LEVEL=1
+  /// APPLAUNCHER_0_PATH=/path/to/level-0
+  /// APPLAUNCHER_0_FRUIT=apple
+  /// PATH=/path/to/level-1:/path/to/level-0
+  /// FRUIT=banana
+  /// \endcode
+  ///
+  /// and calling `environment(0)` would return:
+  ///
+  /// \code
+  /// APPLAUNCHER_LEVEL=0
```

Comment:
I think APPLAUNCHER_LEVEL=0 should not be present (I think the current code, correctly, removes it, to return exactly the same environment as the original system env).